### PR TITLE
Add override image.

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>
https://github.com/stolostron/backlog/issues/25248

### CODE REVIEW

**Comment by xuezhaojun** [comment]
/hold 
Under testing.

**Comment by xuezhaojun** [comment]
/unhold
test passed.

File: pkg/config/agent.go [hunk 1]
```diff
@@ -22,10 +22,15 @@ func GetParsedAgentImage(defaultAgentImageName string) (string, string, string,
 		klog.InfoS("AgentImageName is not set, use default value", "defaultAgentImageName", defaultAgentImageName)
 		AgentImageName = defaultAgentImageName
 	}
-	imgParts := strings.Split(AgentImageName, "/")
+
+	return ParseImage(AgentImageName)
+}
+
+func ParseImage(imageName string) (string, string, string, error) {
+	imgParts := strings.Split(imageName, "/")
```
[the remark this answers is not in the document]
Comment (xuezhaojun) [reply pkg/config/agent.go:30]:
``` go
	imgParts := strings.Split(imageName, "/")
	if len(imgParts) != 2 && len(imgParts) != 3 {
		// image name without registry is also legal.
		return "", "", "", fmt.Errorf("invalid agent image name: %s", imageName)
	}
```
Do we mean the `imgParts` shoudl be 2 or 3?

[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/config/agent.go:30]: in image registry， get image name and tag
```
imageSegments := strings.Split(imageName, "/")
imageNameTag := imageSegments[len(imageSegments)-1]
```

[the remark this answers is not in the document]
Comment (qiujian16) [reply pkg/config/agent.go:30]: should we update here?

[the remark this answers is not in the document]
Comment (xuezhaojun) [reply pkg/config/agent.go:30]: Updated. Remove validation of image length.

**Review by qiujian16 — COMMENTED** [review]
/lgtm

**Pull request merged**